### PR TITLE
[OKD-Core] Do not add maintenance actions if NodeMaintenance CRD is m…

### PR DESCRIPTION
**Pull request by rawagner**
…issing

https://bugzilla.redhat.com/show_bug.cgi?id=1703349

### CODE REVIEW

File: frontend/public/kubevirt/components/node/node.jsx [hunk 1]
```diff
@@ -43,3 +46,31 @@ const NodeStatusWithMaintenance = ({node, maintenances}) => {
   const maintenance = findNodeMaintenance(node, maintenances);
   return maintenance ? <NodeStatus node={node} maintenances={maintenances} TimestampComponent={Timestamp} /> : <StatusIcon status={nodeStatus(node)} />;
 };
+
+const stateToProps = ({k8s}) => ({
+  nodeMaintenanceExists: k8s.getIn(['RESOURCES', 'models', NodeMaintenance.kind]),
```
Comment (suomiy) [inline frontend/public/kubevirt/components/node/node.jsx]: can you add `!!` to be more type safe?

File: frontend/public/kubevirt/components/node/node.jsx [hunk 2]
```diff
@@ -43,3 +46,31 @@ const NodeStatusWithMaintenance = ({node, maintenances}) => {
...
+});
+
+export const NodeDetailsPageWithMaintenance = connect(stateToProps)(({nodeMaintenanceExists, pages, ...rest}) => {
```
Comment (suomiy) [inline frontend/public/kubevirt/components/node/node.jsx]: pages can be in `rest`

File: frontend/public/kubevirt/components/node/node.jsx [hunk 3]
```diff
@@ -43,3 +46,31 @@ const NodeStatusWithMaintenance = ({node, maintenances}) => {
...
+    <DetailsPage
+      {...rest}
+      menuActions={maintenanceMenuActions}
+      pages={pages}
+      resources={resources}
+    />
+  );
+});
+
+const ConnectedNodeMaintenanceKebab = connect(stateToProps)(({nodeMaintenanceExists, actions, kind, resource }) => {
```
Comment (suomiy) [inline frontend/public/kubevirt/components/node/node.jsx]: kind & resource can be in `...rest`

File: frontend/public/components/node.tsx [hunk 4]
```diff
@@ -245,9 +244,7 @@ const pages = [
   pods(({obj}) => <PodsPage showTitle={false} fieldSelector={`spec.nodeName=${obj.metadata.name}`} />),
   events(ResourceEventStream),
 ];
-export const NodesDetailsPage = props => <DetailsPage
+export const NodesDetailsPage = props => <NodeDetailsPageWithMaintenance
   {...props}
```
Comment (suomiy) [inline frontend/public/components/node.tsx]: why not pass menuActions as a prop? Then they would not have to be exported.

File: frontend/public/kubevirt/components/node/node.jsx (hunk 1 above)
Comment (rawagner) [reply frontend/public/kubevirt/components/node/node.jsx]: fixed

File: frontend/public/kubevirt/components/node/node.jsx (hunk 2 above)
Comment (rawagner) [reply frontend/public/kubevirt/components/node/node.jsx]: `pages` are now part of `rest`

File: frontend/public/kubevirt/components/node/node.jsx (hunk 3 above)
Comment (rawagner) [reply frontend/public/kubevirt/components/node/node.jsx]: fixed

File: frontend/public/components/node.tsx (hunk 4 above)
Comment (rawagner) [reply frontend/public/components/node.tsx]: no reason, just forgotten :) `menuActions` are now passed via `props`

**Pull request merged**